drivers/clock_control/clock_control_mcux_ccm_rev3.c

```diff
@@ -0,0 +1,102 @@
+/*
+ * Copyright 2023 NXP
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <errno.h>
+#include <zephyr/sys/util.h>
+#include <zephyr/drivers/clock_control.h>
+#include <zephyr/drivers/clock_control/clock_control_mcux_ccm_rev3.h>
+
+/* Used for driver binding */
+#define DT_DRV_COMPAT nxp_imx_ccm_rev3
+
+extern struct imx_ccm_data mcux_ccm_data;
+extern struct imx_ccm_config mcux_ccm_config;
+
+static int mcux_ccm_on_off(const struct device *dev,
+		clock_control_subsys_t sys, bool on)
+{
+	uint32_t clock_name;
+	const struct imx_ccm_config *cfg;
+
+	clock_name = (uintptr_t)sys;
+	cfg = dev->config;
+
```

Comment:
Is this really `clock_name` here. It sounds to me that is more like `clk_num`.

Reply:
It's called `clock_name` to match the name of the clock cell. However, we can change this to `clk_num` if you think it will add more readability to the driver.